core/arch/arm/kernel/delay.c

```diff
@@ -44,3 +48,37 @@ void mdelay(uint32_t ms)
 {
 	udelay(1000 * ms);
 }
+
+uint64_t utimeout_init(uint32_t us)
+{
+	return read_cntpct() + US2CNT(us);
+}
+
+bool utimeout_elapsed(uint32_t us, uint64_t reference)
```

Comment:
Why give `us` again? Why not just `return read_cntpct() > reference;`? The system counter is guaranteed to start from zero and not to roll-over before 40 years (Arm ARM G5.1.2) so there is no need to consider the wrap-around case.
Then there is no reason to keep the `u` prefix if there is nothing relating to microseconds.
My proposal:
```C
boot timeout_elapsed(uint64_t exp)
{
	return read_cntpct() > exp;
}
```

And since the functions are very simple, they could go in `kernel/delay.h` as `static inline` IMO.

Reply:
looks ok to me. I'll fix.